docs/guides/typescript.md

```diff
@@ -71,6 +71,229 @@ Providing the context and events as generic parameters for the `createMachine()`
 - The event type (`TEvent`) ensures that only specified events (and built-in XState-specific ones) are used in transition configs. The provided event object shapes are also passed on to actions, guards, and services.
 - Events which you send to the machine will be strongly typed, offering you much more confidence in the payload shapes you'll be receiving.
 
+## The VSCode Extension <Badge text="4.26+" />
+
+Using our [VSCode extension](https://marketplace.visualstudio.com/items?itemName=mattpocock.xstate-vscode), you can automatically generate intelligent typings for XState.
+
+Here's how you can get started:
+
+1. Download and install the [VSCode extension](https://marketplace.visualstudio.com/items?itemName=mattpocock.xstate-vscode).
+2. Open a new file and create a typed model using `createModel`. [Want more info on models?](./models.md). For instance:
+
+```ts
+const model = createModel(
+  {
+    value: ''
+  },
+  {
+    events: {
+      FOO: (value: string) => ({ value }),
+      BAR: () => ({})
+    }
+  }
+);
+```
+
+3. Create a machine from the model:
+
+```ts
+const machine = model.createMachine({
+  initial: 'a',
+  states: {
+    a: {
+      on: {
+        FOO: {
+          actions: 'consoleLogValue',
+          target: 'b'
+        }
+      }
+    },
+    b: {
+      entry: 'consoleLogValueAgain'
+    }
+  }
+});
+```
+
+4. Finally, add a 'types' attribute to the machine and save the file:
+
+```ts
+const machine = model.createMachine({
+  types: {},
+  initial: 'a',
+  states: {
+    a: {
+      on: {
+        FOO: {
+          actions: 'consoleLogValue',
+          target: 'b'
+        }
+      }
+    },
+    b: {
+      entry: 'consoleLogValueAgain'
+    }
+  }
+});
+```
+
+5. The extension should automatically add an imported type to the machine:
+
+```ts
+const machine = model.createMachine({
+  types: {} as import('./filename.typegen').Typegen[0],
+  initial: 'a',
+  states: {
+    a: {
+      on: {
+        FOO: {
+          actions: 'consoleLogValue',
+          target: 'b'
+        }
+      }
+    },
+    b: {
+      entry: 'consoleLogValueAgain'
+    }
+  }
+});
+```
+
+6. Add a second parameter into the `createMachine` call - this is where you implement the actions, services, guards and delays for the machine.
+
+```ts
+const machine = model.createMachine(
+  {
+    // Config here
+  },
+  {
+    actions: {
+      consoleLogValue: (context, event) => {
```

Comment:
React-focused recipe would be also much appreciated so people don't get lost on this one